### ISSUE TITLE
nvcc is stale in nvidia/cuda:9.2-cudnn7-runtime-ubuntu18.04

### ISSUE THREAD

**Issue by fischman** [NVIDIA/nvidia-docker#740]
The recently released (#666) `nvidia/cuda:9.2-cudnn7-runtime-ubuntu18.04` image comes with cuda 9.2 per its name, but with nvcc that self-identifies as 9.1. This mismatch breaks the tensorflow build.  Is it intentional (i.e. does tensorflow's build need to be taught that nvcc9.1 can be used with cuda9.2) or does the nvcc binary in the 9.2 image need to be updated?

Repro recipe for nvcc being 9.1:
```
~ $ docker run -it nvidia/cuda:9.2-cudnn7-runtime-ubuntu18.04
root@a41114938ade:/# apt-get update && apt-get install -y --no-install-recommends nvidia-cuda-toolkit
[...]
root@a41114938ade:/# nvcc --version                                                                                                                                                                              
nvcc: NVIDIA (R) Cuda compiler driver                                                                                                                                                                           
Copyright (c) 2005-2017 NVIDIA Corporation
Built on Fri_Nov__3_21:07:56_CDT_2017
Cuda compilation tools, release 9.1, V9.1.85
```

Sample tensorflow build failure:
```
Cuda Configuration Error: CUDA version detected from nvcc (9.1.85) does not match TF_CUDA_VERSION (9.2)
```

**Comment by flx42** [comment]
Hello @fischman 

> The recently released (#666) nvidia/cuda:9.2-cudnn7-runtime-ubuntu18.04 image comes with cuda 9.2 per its name, but with nvcc that self-identifies as 9.1

Not exactly, this image being a `runtime` image, it doesn't come bundled with `nvcc` at all. You install nvcc in your next step, where you do `apt-get install nvidia-cuda-toolkit`. This package comes from Canonical, not NVIDIA:
```
$ apt-cache policy nvidia-cuda-toolkit 
nvidia-cuda-toolkit:
  Installed: (none)
  Candidate: 9.1.85-3ubuntu1
  Version table:
     9.1.85-3ubuntu1 500
        500 http://us.archive.ubuntu.com/ubuntu bionic/multiverse amd64 Packages
```

If you want `nvcc` bundled in an image, use `nvidia/cuda:9.2-cudnn7-devel-ubuntu18.04`. This will be a large image though.
Or you can still use the base image you have, but do `apt-get install cuda-nvcc-9-2` to get nvcc in addition.

You can browse at our list of packages here:
http://developer.download.nvidia.com/compute/cuda/repos/ubuntu1710/x86_64/

**Comment by fischman** [comment]
Perfect, thanks!

FWIW I was thrown by the "1710" in that URL and in general by the presence of 1710 and 16.04 in the 18.04-based https://gitlab.com/nvidia/cuda/blob/ubuntu18.04/9.2/base/Dockerfile
Is nvidia only rebuilding built-for-older-ubuntu-releases packages that turn out to be incompatible with newer ubuntu releases?

**Comment by flx42** [comment]
We qualify this support as "experimental", we are using the unmodified packages of these older distros, they are not being rebuilt. Note that the 17.10 packages were released last week and 17.10 is very similar to 18.04.

Those packages should work just fine on Ubuntu 18.04, but let me know if you find any issue with your tests.

**Comment by flx42** [comment]
Please open a new issue if you have a problem. I hope those images are useful to set the ball rolling with a 18.04 base, even if they are qualified as "experimental".
